File: user-springboot-admin/admin/src/main/java/com/artsail/production/model/domain/HarvestRecord.java
package com.artsail.production.model.domain;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("harvest_record")
public class HarvestRecord implements Serializable {
    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;
    private String recordNo;
    private Long baseId;
    private Long pondId;
    private String species;
    private BigDecimal weight;
    private String unit;
    private String method;
    private String teamName;
    private String operatorName;
    private String status;
    private String remark;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private LocalDateTime harvestTime;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private LocalDateTime createTime;
}

File: user-center-admin/src/types/api/harvest-record.ts
/**
 * 捕捞记录 API 返回的数据结构（契约）
 */

/** 捕捞记录  */
export interface HarvestRecord {
  id: number;
  recordNo: string;
  baseId: number;
  pondId: number;
  species: string;
  weight: number;
  unit: string;
  method: string;
  teamName: string;
  operatorName: string;
  status: string;
  remark: string;
  harvestTime: string;
  createTime: string;
}
